client/source/js/modules/analysis/optimization-ctrl.js

```diff
@@ -2,15 +2,13 @@ define(['./module', 'angular', 'underscore'], function (module, angular, _) {
   'use strict';
 
   module.controller('AnalysisOptimizationController', function ($scope, $http,
-    $interval, meta, cfpLoadingBar, CONFIG, modalService, graphTypeFactory,
+    $interval, meta, cfpLoadingBar, CONFIG, modalService, typeSelector,
     optimizations, optimizationHelpers) {
 
       $scope.chartsForDataExport = [];
 
       $scope.meta = meta;
-      $scope.types = graphTypeFactory.types;
-      // reset graph types every time you come to this page
-      angular.extend($scope.types, angular.copy(CONFIG.GRAPH_TYPES));
+      $scope.types = typeSelector.types;
```

Comment:
yes, I would never be afraid of doing coupling when the concerns are right.

The optimization controller knows it has a children that can `reset` something, is okay to ask for that action when matters. What is not right is that this controller implement the reset. The how should a reset happen is a concern of the guy that holds that state. Makes sense?